### PR TITLE
Update docs to match code.

### PR DESCRIPTION
**Pull request by ADmad**
Refs cakephp/cakephp#14115

### CODE REVIEW

**Comment by BGehrels** [comment]
I guess it's also a good idea to include a bullet point into the [Migration Guide](https://book.cakephp.org/4/en/appendices/4-0-migration-guide.html)

**Comment by BGehrels** [comment]
And if PSR only accepts `string` as `$message`, it might make sense to
- Update the PHP Docs of `\Cake\Log\Log::write` (currently says `mixed`)
- Maybe introduce the type into the method signature of `\Cake\Log\Log::write`
- Update the PHP Docs of `\Cake\Log\LogTrait::log` (currently says `mixed`)
- Maybe introduce the type into the method signature of `\Cake\Log\LogTrait::log`

**Comment by ADmad** [comment]
@BGehrels Done and done :)

File: en/appendices/4-0-migration-guide.rst [hunk 1]
```diff
@@ -345,6 +345,12 @@ Helper
 * The ``safe`` option for ``HtmlHelper::scriptBlock()`` and ``HtmlHelper::scriptStart()``
   has been removed. When enabled it generated ``CDATA`` tags which are only required
   for XHTML which is now defunct.
+  
+Log
+---
+* Logging related methods like ``Cake\Log\LogTrait::log()``, ``Cake\Log\Log::write()`` etc.
+  now only accept string for ``$message`` argument. This change was necessary to align the
+  API with `PSR-3 <https://www.php-fig.org/psr/psr-3/>` standard.
```
Comment (othercorey) [inline en/appendices/4-0-migration-guide.rst]:
```suggestion
  API with `PSR-3 <https://www.php-fig.org/psr/psr-3/>__` standard.
```

**Pull request merged**